data-pipeline/src/stats_exporter.rs

```diff
@@ -0,0 +1,600 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::HashMap,
+    ops::DerefMut,
+    sync::{
+        atomic::{self, AtomicU64},
+        Mutex,
+    },
+    time,
+};
+
+use datadog_trace_normalization::normalize_utils;
+use datadog_trace_protobuf::pb;
+use ddcommon::{connector, tag::Tag, Endpoint};
+use hyper;
+
+const STATS_ENDPOINT_PATH: &str = "/v0.6/stats";
+
+/// The stats saved in the trace exporter are aggregated by BucketKey
+#[derive(Debug, Hash, PartialEq, Eq, Clone)]
+struct AggregationKey {
+    resource_name: String,
+    service_name: String,
+    operation_name: String,
+    span_type: String,
+    http_status_code: u32,
+    is_synthetics_request: bool,
+}
+
+/// The stats stored for each BucketKey
+#[derive(Debug, Default)]
+struct GroupedStats {
+    hits: u64,
+    errors: u64,
+    duration: u64,
+    top_level_hits: u64,
+    ok_summary: datadog_ddsketch::DDSketch,
+    error_summary: datadog_ddsketch::DDSketch,
+}
+
+impl GroupedStats {
+    fn insert(&mut self, span_stats: &SpanStats) {
+        self.duration += span_stats.duration;
+        self.hits += 1;
+
+        if span_stats.is_error {
+            self.errors += 1;
+            let _ = self.error_summary.add(span_stats.duration as f64);
+        } else {
+            let _ = self.ok_summary.add(span_stats.duration as f64);
+        }
+        if span_stats.is_top_level {
+            self.top_level_hits += 1;
+        }
+    }
+}
+
+/// Metadata required in a ClientStatsPayload
```

Comment:
I think the rustdoc comments for public functions and structs should be more robust and follow the patterns we've been using recently in trace-utils and the sidecar. Let me know if you need specific examples, or a hand writing them.